src/firebase-namespace.ts

```diff
@@ -227,6 +246,38 @@ export class FirebaseNamespaceInternals {
       }
     });
   }
+
+  /**
+   * Parse the file pointed to by the FIREBASE_CONFIG_VAR, if it exists.
+   * Or if the FIREBASE_CONFIG_ENV contains a valid JSON object, parse it directly.
+   * If the environmet variable contains a string that starts with '{' it will be parsed as JSON,
+   * otherwise it will be assumed to be pointing to a file.
+   */
+  private loadOptionsFromEnvVar(): FirebaseAppOptions {
+    let config = process.env[FIREBASE_CONFIG_VAR];
+    if (!validator.isNonEmptyString(config)) {
+      return {};
+    }
+    let contents;
+    let jsonContent;
+    try {
+      if (config.startsWith('{')) {
+        // Assume json object.
+        contents = config;
+      } else {
+        // Assume filename.
+        contents = fs.readFileSync(config, 'utf8');
+      }
+      jsonContent = JSON.parse(contents);
```

Comment:
Nit:  Return here and remove the `jsonContent` variable: `return JSON.parse(contents) as FirebaseAppOptions`